### ISSUE TITLE
Expecting a file upload (Status Code: 400)

### ISSUE THREAD

**Issue by Blacx** [BenExile/Dropbox#67]
Dear @BenTheDesigner ,

I'm having an error when running putFile.php.

The error message that appears is as follows:
<code>Catchable fatal error: Object of class stdClass could not be converted to string in C:\xampp\htdocs\dropbox\new\Dropbox\OAuth\Consumer\Curl.php on line 103</code>

But I tried to change <code>$response['body']->error</code> on  \Dropbox\OAuth\Consumer\Curl.php file at line 103 be as follows:
<code>$message = $response['body']->error->file . ' (Status Code: ' . $response['code'] . ')';</code>

And the exception that appears now is "<b>Expecting a file upload</b>".

Here is the putFile.php which I modified:

<pre>
require_once('bootstrap.php');
try {
    $tmp = tempnam('/tmp', 'dropbox');
    $data = 'This file was uploaded using the Dropbox API!';
    file_put_contents($tmp, $data);    
    $put = $dropbox->putFile($tmp, 'api_upload_test.txt');
    unlink($tmp);   
    var_dump($put); 
} catch (\Dropbox\Exception $e) {
    var_dump($e);
}
</pre>

Here is the output:
![Expecting a file upload](https://f.cloud.github.com/assets/1876100/60296/e9ed0852-5c06-11e2-87c7-61cfd082e74f.png)

**Comment by BenExile** [comment]
Hi @Blacx 

The fatal error is due to Dropbox returning an object as the error message. I have submitted a ticket to Dropbox as `$response['body']->error`, to my knowledge, has always been a string. I'll need to wait for their response before I can reliably fix this issue (it might be a problem on their end).

The Exception thrown suggests no file was uploaded but I have not seen this since the very early days of testing the API and getting cURL to prepare the POST request as required. Is this problem apparent if you use a 'permanent' file that is on your system?

I spent half an hour on my lunch break trying to replicate this issue but no luck so far. @henrylevak (sorry to drag you in) - As you did have the same problem with the previous issue are you able to confirm if this problem also affects you?

**Comment by henrylevak** [comment]
@Blacx I'll test it when I get home tonight - I was able to successfully upload files earlier this week - but I'll try it later tonight as well.

**Comment by Blacx** [comment]
@BenTheDesigner :
Yes, I am having the same problem when I enter the address of the file directly as a parameter of putFile () method. If it is the intention of 'permanent' you mean Ben.

ex: <code>$put = $dropbox->putFile ('C:\directfile.txt', 'api_upload_test.txt');</code>

@henrylevak : I hope it help me.

**Comment by BenExile** [comment]
I do have confirmation from Dropbox that the API returns error messages inconsistently... Useful! This is being looked at by their team, but I'm told the likelihood is that this behaviour won't be changed in version 1 of the API.

I'm still unable to replicate the error with uploading files but have addressed the code that was triggering the fatal in commit 5bbf95b5b36f4da9d43a52c8c96e9b20d2759368.

@henrylevak - Thanks I'll await your reply.

**Comment by henrylevak** [comment]
Sorry to disappoint all but I haven't had any time to check on this. I had to finish a feature last night and ran out of time . I'll try again tonight :(

**Comment by BenExile** [comment]
No problem! @Blacx - Can you advise if this problem occurs for all uploads or just some of them?

**Comment by henrylevak** [comment]
So I used @Blacx 's code and had no problems what so ever..

<?php

require_once('bootstrap.php');
try {
    $tmp = tempnam('/tmp', 'dropbox');
    $data = 'This file was uploaded using the Dropbox API!!!!!';
    file_put_contents($tmp, $data);  
    $put = $dropbox->putFile($tmp, 'api_upload_test1.txt');
    unlink($tmp);  
    var_dump($put); 
} catch (\Dropbox\Exception $e) {
    var_dump($e);
}

This uploaded the file perfectly...

**Comment by BenExile** [comment]
Same for me @henrylevak. I'll need to get a 5.3.1 install up and running to see if I can replicate.

**Comment by Blacx** [comment]
Hi @BenTheDesigner only <b>putFile.php</b> is problematic for me, the other work normally even <i>putStream.php</i> also work well.
I would be very delighted if you intend to try it in version 5.3.1 ben. Thank you very much in advance. 

And thank you @henrylevak to try and provide information.

**Comment by BenExile** [comment]
Hi @Blacx 

I'm unable to replicate this issue under 5.3.1. It's very strange that this only affects the putFile method, too. Can you advise if the problem persists and if you've found out any more about the issue?

**Comment by Blacx** [comment]
Hi @BenTheDesigner, 

I was trying to execute putFile in PHP 5.3.18 and the result goes perfectly. But it's weird and still not be another clue to run it in PHP 5.3.1.

<i>Still trying to solve the problem while waiting for a solution.</i>

**Comment by BenExile** [comment]
Hi @Blacx,

Unfortunately I'm still unable to replicate this. Do you have any more information about what might be causing the issue?

**Comment by Blacx** [comment]
Hi @BenTheDesigner,

I do not have a solution as well. While I still think that the error occurred because of PHP version 5.3.1 that I use.

**Comment by BenExile** [comment]
PHP 5.3.1 does meet the minimum requirement and all code to date is tested under that version. I'll leave this issue open so others can comment if they experience the same issue. Sorry I can't help more on this occasion.

**Comment by applemate** [comment]
I was running php 5.6.2

**Comment by abusohib** [comment]
I faced same problem. i was running php 5.6.10

I found the answer . There is a new Variable included with curl in PHP 5.5: CURLOPT_SAFE_UPLOAD this is set to false by default in PHP 5.5 and is switched to a default of true in PHP 5.6.

I have set CURLOPT_SAFE_UPLOAD = false in https://github.com/BenExile/Dropbox/commit/79ea42dcc4afd4e21af0597ff0c6e47c2a557e21

It has fixed my problem.

**Comment by reimax** [comment]
not work.  Expecting a file upload ...
